src/db.c

```diff
@@ -1630,7 +1635,7 @@ int getKeysFromCommand(struct redisCommand *cmd, robj **argv, int argc, getKeysR
     } else if (!(cmd->flags & CMD_MODULE) && cmd->getkeys_proc) {
         return cmd->getkeys_proc(cmd,argv,argc,result);
     } else {
-        return getKeysUsingCommandTable(cmd,argv,argc,result);
+        return getKeysUsingLegacyRangeSpec(cmd,argv,argc,result);
```

Comment:
we plan to do it at some point, but we think the complexity of getting key indices in a generic way is complicated both to code and both in CPU
also there the ambiguity problem, for example if a user has a keyname called "streams" in XREAD

Reply:
Isn't this specific issue a limitation we should try to address with the keyspec?

Reply:
It would be nice, but I don't have an idea. 
The only way I can think of is to add a full argument spec for the command (like the one in commands.json in redis.io), but that would be complicated for client libraries to work with. 
This simple approach is easy to use and will almost always work. 
Additionally we aim to gradually depreciate all these keyword argument commands. 

Meanwhile, since redis already has these getkeys procs, I think is should keep using them

Reply:
I'm OK with leaving the Redis implementation as is but I have a concern about the keyspec. Basically this means we should not use `keyword` for `XREAD` because clients implementing it may parse the command the wrong way.

Reply:
but what else can we do at this point in time?
GEORADIUS* XREAD* SORT*, STRALGO, MIGRATE and MEMORY are terrible in that manner (hard to find the key names).

Reply:
@oranagra I did mention `limit` and `incomplete` specs in the top comment, but I did miss the fact that some `incomplete` specs can be completely empty (i.e. UNKNOWN begin_search) which should tell the client that COMMAND GETKEYS must be used

Reply:
@guybe7 i meant a limit for the keyword search, not for the range spec

Reply:
just summarizing, for the record:
XREAD is actually not an issue: assuming the syntax is correct, there are no arbitrary strings in argv up until streams (only command name, optional tokens, and numbers), so it's safe to use "keyword"

the command we have issues with are MIGRATE, STRALGO, and SORT
for MIGRATE, because the token KEYS, if exist, must be the last token, we can search in reverse. it one of the keys is actually the string "keys" will return just a subset of the keys (hence, it's "incomplete")
for SORT and STRALGO we can use this heuristic (the keys can be anywhere in the command) and therefore we added a key spec that is both "incomplete" and of "unknown type"

if a client encounters an "incomplete" spec it means that it must find a different way (either COMMAND GETKEYS or have its own parser) to retrieve the keys.
please note that all commands, apart from the three mentioned above, have "complete" key specs

Reply:
This is a good description, maybe it should be copied to the top comment, or to a redis-doc PR. 
For now I was thinking of not suggesting to make a redis-doc PR since we currently added this data to the COMMAND command, but we plan on moving it out of it soon and into the COMMANDS command.